src/main/java/com/aws/greengrass/lifecyclemanager/GreengrassService.java

```diff
@@ -84,6 +86,8 @@ public class GreengrassService implements InjectionActions {
 
     // dependencies that are explicitly declared by customer in config store.
     private final Topic externalDependenciesTopic;
+    @Getter @Setter
+    private Watcher externalDependenciesTopicWatcher;
```

Comment:
Will there always be just one watcher?

Reply:
There could be more, but removing just the one watcher which is registered when the service is initialized should be sufficient

Reply:
There could be more watchers for dependencies? and if so, why is removing just one sufficient?

Reply:
I have not come across a situation where there were more than one watchers for a service dependency topic, I did not exhaustively look at situations on this. However, it seemed like the watcher which was registered during the initial watcher registration still being alive when services were closed was causing all the problems. I also wanted to make minimal changes on this.

Reply:
But if there is a recommendation that the entire Watchers set should be emptied when a service is being closed then that can be done as well